flycheck.el

```diff
@@ -3051,7 +3068,17 @@ Intended for use with `next-error-function'."
   (let ((pos (flycheck-next-error-pos n reset)))
     (if pos
         (goto-char pos)
-      (user-error "No more Flycheck errors"))))
+      (let ((result (cl-remove-if
```

Comment:
This implements the "no more errors, so let's see if there are errors from another buffer" logic.